packages/core/src/internal/decorators/animate.ts

```diff
@@ -12,25 +12,39 @@ import {
 } from '../motion/interfaces.js';
 import { runPropertyAnimations } from '../motion/utils.js';
 
-// https://www.typescriptlang.org/docs/handbook/decorators.html#class-decorators
+// decorator factory that extends the component constructor to inject animations code into it
 export function animate(config: PropertyDrivenAnimation) {
   // eslint-disable-next-line @typescript-eslint/ban-types
-  return function _DecoratorName<T extends { new (...args: any[]): {} }>(constr: T) {
-    const _constr = constr as any;
-    return (class extends _constr {
-      _animations: any;
+  return function _animationDecorator<T extends { new (...args: any[]): AnimatableElement }>(constructor: T) {
+    return (class extends constructor {
+      private _animationReady = false;
+
+      private _animationDemoMode = false;
+
+      _animations = config;
 
       updated(props: Map<string, any>) {
         super.updated(props);
 
-        const self = (this as unknown) as AnimatableElement & { _animations: PropertyDrivenAnimation };
-        self._animations = config || void 0;
+        // const self = (this as unknown) as AnimatableElement;
+        // self._animations = config || void 0;
 
-        if (!self.hasAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME)) {
-          self.setAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME, AnimationStatus.ready);
-        } else if (!self.hasAttribute('_demo-mode')) {
-          // ignore if element has the _demo-mode attribute set; _demo-mode is used for docs, static examples, and stories
-          runPropertyAnimations(props, self);
+        // init-style booleans here to skip need for expensive attr lookup
+        if (!this._animationReady && !this.hasAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME)) {
```

Comment:
Conditional complexity is due to guards that simplify/reduce the number of lookups the decorator needs to do.